### DISCUSSION TITLE
Translation updates ?

### DISCUSSION THREAD

**Discussion by Harvester57** [openstreetmap/iD#9478] [Q&A]
Hi,

It could be a very dumb question, so apologies in advance if I'm missing something obvious, but how and when are translations supposed to be updated in the GUI (either on the main OSM website, or in the dev editor at https://ideditor.netlify.app/) ?

According to the documentation (https://github.com/openstreetmap/iD/blob/develop/CONTRIBUTING.md#translating) it is supposed to be synced when a new commit is pushed to the `dev` branch, but I have been adding translations to Transifex for two months now, and I haven't seen the new strings that I know for sure I updated. 

To give a concrete example, betwenn the releases 2.23.0 and 2.24.0, all the translations related to crossings were lost (?) for `crossing=xxx` field, mean,ing that the `crossing=marked` preset is not shown anymore when you add a new point and search to add a new crossing there. At first, I tought that I would fix it on Transifex, by searching for missing translations related to crossings, and indeed some strings were erased by the Transifex bot around the time it disapared from the iD editor. So far so good, I added the strings back, and waited for the next commit on the dev branch to check again on the dev editor if it was showing up.

While there have been several new commits pushed to the dev branch, nothing has changed. The new 2.24 and 2.24.1 releases have been published since then, and even now ther hasn't been any sync to the translations, so I guess I must be missing something... Is the language on Transifex supposed to be marked as "Ready to use" to be synced ? Or is the sync mecahnism broken ?

To give concrete example of the crossing translation problem, here is a screenshot from the main iD editor website:

![image](https://user-images.githubusercontent.com/7060295/215265703-1d6bc448-0957-4b4b-a956-86c291cbb659.png)

As you can see, only 3 subtypes of crossings are showing with a description, and they are the only 3 types that are shown when searching in the search box.

Here is the same object, viewed from the dev iD editor:

![image](https://user-images.githubusercontent.com/7060295/215265844-79f6437a-7167-42fc-94e5-b7091bffd2c0.png)

(The node is here by the way: https://www.openstreetmap.org/node/1587375113)

**Comment by LilianaMoore** [comment]
I haven't quite figured it out yet, but it seems to me that some of the texts used by the ID editor do not come from transifex translations, but are fetched directly from various other translation sources such as:
- translated OSM Wiki documentation pages,
- Data items https://wiki.openstreetmap.org/wiki/Data_items
- taginfo https://wiki.openstreetmap.org/wiki/Taginfo
- and perhaps others

It's hard to know how it's all connected in the ID editor because I haven't been able to get to specific documentation for translators regarding this matter.
Edit: I can confirm that my latest translations from transifex are being fetched by dev preview site https://ideditor.netlify.app/ without any problems.

**Reply by tordans** [reply]
The part of the docs that explains Translations is https://github.com/openstreetmap/iD/blob/develop/CONTRIBUTING.md#translating

I think it would be good improve the structure of that part a bit to clarify which areas of the app gets data from where. And the Data Items should also be added.

**Reply by etienneJr** [reply]
Hi,
I have the same issue with translations changed in transifex 2 months ago, and still not in the dev editor.
The translation paragraph of the contributing readme seems almost identical to the one from 1 year ago, so I don't think it has been clarified ?!?
[edit] do I need to create an issue to track this request ?
